bundles/org.openhab.binding.fenecon/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,144 @@
+	<channel-type id="lastUpdate">
+		<item-type>DateTime</item-type>
+		<label>Last Update</label>
+		<description>Last successful update via REST-API from the FENECON system</description>
+		<category>Time</category>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="essSoc">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Battery State</label>
+		<description>Battery state of charge in percent</description>
+		<category>BatteryLevel</category>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+	<channel-type id="chargerPower">
+		<item-type>Number:Power</item-type>
+		<label>Charger Power</label>
+		<description>Current charger power of energy storage system in watt.</description>
+		<category>Energy</category>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+	<channel-type id="dischargerPower">
+		<item-type>Number:Power</item-type>
+		<label>Discharger Power</label>
+		<description>Current discharger power of energy storage system in watt.</description>
+		<category>Energy</category>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+	<channel-type id="emergencyPowerMode">
+		<item-type>Switch</item-type>
+		<label>Emergency Power Mode</label>
+		<description>Indicates if there is grid power is off and the emergency power mode is on.</description>
+		<category>Switch</category>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="productionActivePower">
+		<item-type>Number:Power</item-type>
+		<label>Producer Power</label>
+		<description>Current active power producer load in watt.</description>
+		<category>Energy</category>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+	<channel-type id="sellToGridPower">
+		<item-type>Number:Power</item-type>
+		<label>Sell Grid Power</label>
+		<description>Current sell to grid exchange power in watt.</description>
+		<category>Energy</category>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+	<channel-type id="sellToGridEnergy">
+		<item-type>Number:Energy</item-type>
+		<label>Sell Grid Energy</label>
+		<description>Total energy exported to the grid in watt per hour.</description>
+		<category>Energy</category>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+	<channel-type id="consumptionActivePower">
+		<item-type>Number:Power</item-type>
+		<label>Consumer Power</label>
+		<description>Current active power consumer load in watt.</description>
+		<category>Energy</category>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+	<channel-type id="buyFromGridPower">
+		<item-type>Number:Power</item-type>
+		<label>Buy Grid Power</label>
+		<description>Current buy from grid exchange power in watt.</description>
+		<category>Energy</category>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+	<channel-type id="buyFromGridEnergy">
+		<item-type>Number:Energy</item-type>
+		<label>Buy Grid Energy</label>
```

Comment:
```suggestion
		<label>Bought Grid Energy</label>
```

Reply:
More common is import and or export. I'm not a native english speaker, but these labels might need some improvement. Maybe look at the Tibber binding that has similar channels.

Reply:
Done